src/compile/data/stack.ts

```diff
@@ -206,12 +207,12 @@ export class StackNode extends DataFlowNode {
       }
       return [vgField(dimensionFieldDef)];
     }
-    return [];
+    return groupby || [];
   }
 
   public assemble(): VgTransform[] {
     const transform: VgTransform[] = [];
-    const {facetby, dimensionFieldDef, stackField: field, stackby, sort, offset, impute, groupby, as} = this._stack;
+    const {facetby, dimensionFieldDef, stackField: field, stackby, sort, offset, impute, as} = this._stack;
```

Comment:
Note: this commit changes the behavior of `getGroupbyFields`. 
An alternative would be to leave `getGroupbyFields` unchanged and instead do:
`const {facetby, dimensionFieldDef, stackField: field, stackby, sort, offset, impute, groupby=[], as} = this._stack;`
and then further down do `groupby: this.getGroupbyFields().concat(groupby, facetby),`